src/Compiler/Service/ServiceDeclarationLists.fs

```diff
@@ -1144,22 +1145,74 @@ type DeclarationListInfo(declarations: DeclarationListItem[], isForType: bool, i
+                    // but still merging extension method if it tries to shadow other stuff than a Property
+                    // we proceed with a pre-scan to see if we hit the particular case, in which case we partition
+                    // items to be split and those that were initially remaining grouped.
+                    // If we don't hit the specific case, or have a single entry, we keep the same logic as originally
+                    // N.B: due to the logic returning 1 to N instead of 1, the next stage of the pipeline is List.concat
+                    // introduced for this RFC
+                    let hasBothPropertiesAndExtensionMethods items =
+                        let rec inner hasProperty hasExtensionMethod items =
+                            if hasProperty && hasExtensionMethod then
+                                true
+                            else
+                                match items with
+                                | [] -> hasProperty && hasExtensionMethod
+                                | item :: tail when item.Kind = CompletionItemKind.Property                 -> inner true hasExtensionMethod tail 
+                                | item :: tail when item.Kind = CompletionItemKind.Method(isExtension=true) -> inner hasProperty true tail
+                                | _ :: tail                                                                 -> inner hasProperty hasExtensionMethod tail
+                        inner false false items
+                    function
+                        | _, ([_] as items) 
+                        | _,items when not (hasBothPropertiesAndExtensionMethods items) ->
+                            let item = items.Head
+                            [textInDeclList item, textInCode item, items]
+                        | _, items (* RFC-1137 we have both property and extension method ...*) ->
+                            let toSplit, together =
+                                items
+                                |> List.partition
+                                    (fun item ->
+                                        match item.Kind with
+                                        | CompletionItemKind.Property | CompletionItemKind.Method(isExtension=true) -> true
+                                        | _ -> false
+                                    )
+                            [
+                                let rec createSublists list =
+                                    match list with
+                                    | [] -> []
+                                    | _::tail -> list :: createSublists tail
+                                for items in createSublists toSplit do
+                                    let item = items.Head
+                                    textInDeclList item, textInCode item, items
+                                if not together.IsEmpty then
```

Comment:
I think we could technically skip this, it would always show up items that can't be called anyways (so long there is a property, in current version of the language, you can't call anything else).

It will be empty in most cases, so it will not impact performance unless we are in some strange edge case.